src/smartdc/lib/smartos_prompt_config.sh

```diff
@@ -1339,6 +1438,51 @@ sed -e "s|^root:[^\:]*:|root:${root_shadow}:|" /etc/shadow > /usbkey/shadow \
 
 cp -rp /etc/ssh /usbkey/ssh || fatal "failed to set up preserve host keys"
 
+if [ $boot_non_removable == "yes" ]; then
+    printf "%-56s" "Creating self-bootable $BOOTPOOL pool... "
+
+    # Reality check $BOOTPOOL was created with -B.
+    # Easiest way to do this is to check for the `bootsize` property not
+    # its default, which is NO bootsize.
+    zpool get bootsize ${BOOTPOOL} | grep -q -w default
+    if [[ $? -eq 0 ]]; then
+	fatal "\nDesired bootable pool $BOOTPOOL was not created with -B"
+    fi
+
+    # Create BOOTPOOL/boot and set bootfs.
+    zfs create ${BOOTPOOL}/boot || fatal "Cannot create boot filesystem"
```

Comment:
We might want to explicitly add `-o encryption=off` here. That way it doesn't need to care about the rest of the pool.

Reply:
Good idea.  Will update later today.

Reply:
Check now.